Utils/distroutils.py

```diff
@@ -0,0 +1,499 @@
+import os
```

Comment:
Am I correct in that you wrote this file from scratch?  Impressive.  There is a lot of code here to handle the various cases.  WALinuxAgent does most (all?) of what you want, and supports more distros and distro versions.  If possible, consider using that code instead.  Or, did you already do that and created this file from pieces of the agent?

How are you testing this code?

Reply:
Most of the code is borrowed from WALinuxAgent, with a few tweaks like class initialization, refactoring, and ensuring python 3.6 and 3.8 compatibility, and making sure the subprocess is used with shell=False.

The other distros you mentioned inherited the functions that VMAccess uses form the AbstractDistro class. So there was no need to implement them here.